### ISSUE TITLE
[java] False positive for useless parenthesis

### ISSUE THREAD

**Issue by mnlipp** [pmd/pmd#1023]
**Affects PMD Version:** 6.2.0

**Rule:** UselessParenthesis

**Description:** See example, they aren't useless. First character in `str1` may well be an underscore.

**Code Sample demonstrating the issue:**

```
public class Test {

	private String str1;
	
	public String themeName() {
		return (Character.toUpperCase(str1.charAt(0))
				+ str1.substring(1)).replace('_', ' ');
	}

}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]* eclipse-pmd 1.11
